test/api/teaching_period_test.rb

```diff
@@ -28,6 +28,23 @@ def test_get_teaching_periods
     end
   end
 
+  def test_get_teaching_periods_details
+    expected_tp = TeachingPeriod.second
+
+    # perform the GET 
+    get with_auth_token "/api/teaching_periods/#{expected_tp.id}"
+    actual_tp = last_response_body
+    assert_equal 200, last_response.status
+    
+    # Check the returned details match as expected 
+    assert_equal actual_tp['period'], expected_tp.period
```

Comment:
There is a helper that can simplify this for you - check `assert_json_matches_model`